src/System.CommandLine.Tests/Invocation/CancelOnProcessTerminationTests.cs

```diff
@@ -15,222 +15,76 @@ namespace System.CommandLine.Tests.Invocation
-        public async Task CancelOnProcessTermination_provides_CancellationToken_that_signals_termination_when_no_timeout_is_specified(int signo)
+        public enum Signals
         {
-            const string ChildProcessWaiting = "Waiting for the command to be cancelled";
-            const int CancelledExitCode = 42;
-
-            Func<string[], Task<int>> childProgram = (string[] args) =>
-            {
-                var command = new Command("the-command");
-            
-                command.SetHandler(async context =>
-                {
-                    var cancellationToken = context.GetCancellationToken();
-
-                    try
-                    {
-                        context.Console.WriteLine(ChildProcessWaiting);
-                        await Task.Delay(int.MaxValue, cancellationToken);
-                        context.ExitCode = 1;
-                    }
-                    catch (OperationCanceledException)
-                    {
-                        // For Process.Exit handling the event must remain blocked as long as the
-                        // command is executed.
-                        // We are currently blocking that event because CancellationTokenSource.Cancel
-                        // is called from the event handler.
-                        // We'll do an async Yield now. This means the Cancel call will return
-                        // and we're no longer actively blocking the event.
-                        // The event handler is responsible to continue blocking until the command
-                        // has finished executing. If it doesn't we won't get the CancelledExitCode.
-                        await Task.Yield();
-
-                        context.ExitCode = CancelledExitCode;
-                    }
-
-                });
-
-                return new CommandLineBuilder(new RootCommand
-                       {
-                           command
-                       })
-                       .CancelOnProcessTermination()
-                       .Build()
-                       .InvokeAsync("the-command");
-            };
-
-            using RemoteExecution program = RemoteExecutor.Execute(childProgram, psi: new ProcessStartInfo { RedirectStandardOutput = true });
-            
-            Process process = program.Process;
-
-            // Wait for the child to be in the command handler.
-            string childState = await process.StandardOutput.ReadLineAsync();
-            childState.Should().Be(ChildProcessWaiting);
-
-            // Request termination
-            kill(process.Id, signo).Should().Be(0);
+            SIGINT = 2, // Console.CancelKeyPress
+            SIGTERM = 15 // AppDomain.CurrentDomain.ProcessExit
+        }
 
-            // Verify the process terminates timely
-            bool processExited = process.WaitForExit(10000);
-            if (!processExited)
+        [Fact]
```

Comment:
Any reason for using the runtime platform check rather than making a `LinuxOnlyFact`?

Reply:
My initial intention was to enable it on macOS too, but then it turned out that `RemoteExecutor` does not support this scenario, so instead of adding new attribute (This repo defines only `LinuxOnlyTheory` attribute. ) I've added the ugly if statement.